### PR TITLE
Add support for NVIDIA GeForce GTX 1080, perhaps others, v. 2

### LINKED ISSUE #9: Supports proprietary driver installation for ATI/AMD, but not NVIDIA
The project has `steamos_ati`, but no `steamos_nvidia`.  Would be nice to have both.

### PR DESCRIPTION
**Pull request by jcorrado**
replaces PR #11
fixes #9

### CODE REVIEW

File: util/steambox [hunk 1]
```diff
@@ -51,9 +51,14 @@ declare -a DRIDEVS=()
 if [[ -d /dev/dri ]] ; then
 	DRIDEVS+=(/dev/dri/*)
 fi
-if [[ -d /dev/ati ]] ; then 
+if [[ -d /dev/ati ]] ; then
 	DRIDEVS+=(/dev/ati/*)
 fi
+for cdev in nvidiactl nvidia0 nvidia-modeset; do
```
Comment (rsharo) [inline util/steambox]: @jcorrado My one thought on this: if someone is running two graphics cards (for example, an SLI build), will they possibly have `/dev/nvidia1` as well?

The possibility definitely won't stop me from merging this, but I wonder if we should say `nvidia[0-9]`

File: steambox_nvidia/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM steambox
+MAINTAINER Jereme Corrado "jereme@zoion.net"
+
+USER root
+
+RUN apt-get install -y debian-archive-keyring
+RUN echo "deb http://ftp.debian.org/debian/ jessie-backports main contrib non-free" > /etc/apt/sources.list.d/jessie-backports.list
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive \
+    apt-get install -y -t jessie-backports nvidia-driver
```
Comment (rsharo) [inline steambox_nvidia/Dockerfile:9]: @jcorrado I still have the concern that this is installing a Debian-specific driver.  It may work ok for people running vanilla Debian host machines (like you and me :) ), but I'm not as sure about people running Ubuntu, Fedora, Mint, etc.

I'm going to accept this as-is though because having  Debian+Nvidia support is better than having no Nvidia support at all. :)

It can always be addressed in a follow-on changeset if anyone has an issue.

**Review by rsharo — APPROVED** [review]
@jcorrado I've approved this PR, but I'm going to hold off until tomorrow to merge it in case you have comments.

Wanted to let you know, though... you don't have to close a PR and open a new one to address comments or make changes.  If you make changes on the PR branch (ISSUE-2 for PR #11, master for PR #12 ), they will automatically be posted to the PR.

You can change the code and it'll auto-post to the PR right up to the time of merge.

File: util/steambox (hunk 1 above)
Comment (jcorrado) [reply util/steambox]: @rsharo I think a match on additional `nvidia` nodes makes good sense.  And thank you for the advice on updating a PR; I was just this morning thinking it very clunky to have to close and open additional requests to follow up on commentary.

I will push this update shortly.

File: steambox_nvidia/Dockerfile (hunk 2 above)
Comment (jcorrado) [reply steambox_nvidia/Dockerfile:9]: I'm going to be exploring the use of Valve's Nvidia driver, as you proposed.  I think we can view this initial strategy as an interim solution.

File: util/steambox (hunk 1 above)
Comment (jcorrado) [reply util/steambox]: Pushed.

Comment (jcorrado) [reply util/steambox]: Reverted - I've overlooked something.

Comment (jcorrado) [reply util/steambox]: Pushed anew.  It's a little verbose, but I wanted to preserve the character device check.

**Comment by rsharo** [comment]
@jcorrado Merged! Feel free to open new PRs if you find you want to change anything later.

Thanks!

File: steambox_nvidia/Dockerfile (hunk 2 above)
Comment (rsharo) [reply steambox_nvidia/Dockerfile:9]: @jcorrado Sounds good.  Some observations below that may help in your experimentation.  Please bear with me if I'm saying things you already know.

If it is true that the docker image's driver must be identical to the host's driver, then it may be challenging to use SteamOS' driver in many cases.  While the docker image can have different shared objects, binaries, and configuration files, it will always use the *same* kernel as the host.  This means that a user would have to (re)compile the SteamOS driver for their host distro's kernel, which may be a different version and have a different configuration than what Valve expected.

Now, for you and me, this will (probably) not be a problem -- SteamOS Brewmaster is based on Debian Jessie.  My host is also Debian Jessie, so I'm using the kernel version that SteamOS expects.  But just to be sure I didn't have problems, I opted to use the proprietary ATI driver because I knew I could (DKMS) build that driver for both image and host regardless of the kernel version:  using the proprietary driver prevented me from having to port host driver to SteamOS or vice versa.

The tradeoff was the weird build process you see in `steambox_ati`, though:  First `steambox_ati/Dockerfile.1` runs to prepare an image for driver install, then the Makefile walks through interactive install and commits the result, then `steambox_ati/Dockerfile` runs to finalize the image.  It is ugly, but it works... :-/

So, in a nutshell, if your kernel version and/or config deviates significantly from SteamOS, then it might be hard to make host and image drivers match.  I still don't know if I believe this will cause a problem, since the docker container is unable to load a driver anyway -- it will just use the one that the host installed.

Then there's that pesky matter of build instructions.  The readme tells people how to build the images, but it doesn't tell them what to do to their host machine.  

Overall, it may be challenging to maintain a portable set of Dockerfiles that also include 3D drivers.  I guess we'll know more as people use this code and possibly complain or ask for help.
